src/treemodel/selection.js

```diff
@@ -0,0 +1,261 @@
+			 *
+			 * @private
+			 * @property {Array.<LiveRange>}
+			 */
+			this._ranges = [];
+
+			/**
+			 * Specifies whether the last added range was added as a backward or forward range.
+			 *
+			 * @private
+			 * @property {Boolean}
+			 */
+			this._lastRangeBackward = false;
+		}
+
+		/**
+		 * Selection anchor. Anchor may be described as a position where the selection starts.
+		 * Together with {@link #focus} they define the direction of selection, which is important
+		 * when expanding/shrinking selection. When there are no ranges in selection anchor is null.
+		 * Anchor is always a start or end of the most recent added range. It may be a bit unintuitive when
+		 * there are multiple ranges in selection.
+		 *
+		 * @property {treeModel.LivePosition|null}
+		 */
+		get anchor() {
+			if ( this._ranges.length > 0 ) {
+				let range = this._ranges[ this._ranges.length - 1 ];
+
+				return this._lastRangeBackward ? range.end : range.start;
+			}
+
+			return null;
+		}
+
+		/**
+		 * Selection focus. Focus is a position where the selection ends. When there are no ranges in selection,
+		 * focus is null.
+		 *
+		 * @see {#anchor}
+		 * @property {treeModel.LivePosition|null}
+		 */
+		get focus() {
+			if ( this._ranges.length > 0 ) {
+				let range = this._ranges[ this._ranges.length - 1 ];
+
+				return this._lastRangeBackward ? range.start : range.end;
+			}
+
+			return null;
+		}
+
+		/**
+		 * Returns whether the selection is collapsed. Selection is collapsed when all it's ranges are collapsed.
+		 *
+		 * @property {Boolean}
+		 */
+		get isCollapsed() {
+			for ( let i = 0; i < this._ranges.length; i++ ) {
+				if ( !this._ranges[ i ].isCollapsed ) {
+					return false;
+				}
+			}
+
+			return true;
+		}
+
+		/**
+		 * Adds a range to the selection. Added range is copied and converted to {@link treeModel.LiveRange}. This means
+		 * that passed range is not saved in the Selection instance and you can safely operate on it. Accepts a flag
+		 * describing in which way the selection is made - passed range might be selected from {@link treeModel.Range#start}
+		 * to {@link treeModel.Range#end} or from {@link treeModel.Range#start} to {@link treeModel.Range#end}. The flag
+		 * is used to set {@link #anchor} and {@link #focus} properties.
+		 *
+		 * @param {treeModel.Range} range Range to add.
+		 * @param {Boolean} [isBackward] Flag describing if added range was selected forward - from start to end (`false`)
+		 * or backward - from end to start (`true`). Defaults to `false`.
+		 */
+		addRange( range, isBackward ) {
+			pushRange.call( this, range );
+			this._lastRangeBackward = !!isBackward;
+
+			this.fire( 'update' );
+		}
+
+		/**
+		 * Unbinds all events previously bound by this selection, including events bound by created {@link treeModel.LiveRange}s.
+		 */
+		detach() {
+			detachRanges.call( this );
+		}
+
+		/**
+		 * @see {@link treeModel.AttributeList#getAttr}
+		 */
+		getAttr( key ) {
```

Comment:
So it seems that it requires refactoring. Those method are duplicated right now in `Node` and `Selection` only for one reason – because initially we had them directly in `Node` because attribute list was integral part of it. Now it's not, so all those methods are unnecessary.